src/dpl/src/Place.cpp

```diff
@@ -841,14 +836,29 @@ PixelPt Opendp::diamondSearch(const Cell* cell,
                           y_max,
                           x_offset,
                           y_offset,
-                          best_pt,
-                          best_dist);
+                          cur_pt,
+                          cur_dist);
+
+        if (cur_pt.pixel && cur_dist < best_dist) {
+          best_pt = cur_pt;
+          best_dist = cur_dist;
+        }
+
+        // Mark the current position as visited
+        visitedPositions.insert({x_offset, y_offset});
+
+        // Enqueue neighboring positions for exploration
+        positionsQueue.push({x_offset - 1, y_offset});
+        positionsQueue.push({x_offset + 1, y_offset});
+        positionsQueue.push({x_offset, y_offset - 1});
+        positionsQueue.push({x_offset, y_offset + 1});
```

Comment:
rather than queue and later skipping you could keep a collection of enqued points and skip adding redundant ones here.  Checking should be cheaper once using a heap.